data/addFieldsToCollection.js

```diff
@@ -88,6 +109,77 @@ async function setLevelInCourses() {
   });
 }
 
+/* 
+  Script to calculate the post reqs for each sis course.
+*/
+async function setPostReqsInSISCourses() {
+  await db.connect();
+  var counter = 0;
+  SISCV.find({ version: { $elemMatch: { postReq: { $size: 0 } } } }).then(
+    async (res) => {
+      for (let course of res) {
+        let postReqs = [];
+        await SISCV.find({
+          versions: {
+            $elemMatch: {
+              preReq: { $elemMatch: { Expression: { $regex: course.number } } },
+            },
+          },
```

Comment:
Does it make sense to match any version, or only versions AFTER the semester of the version? 

Like say we have a version of Gateway Java (Fall 2022). We want the version of IP that comes AFTER Fall 2022. Are we doing this rn? What if prerequisites change between semesters?

Reply:
I might be wrong, but I don't think we can query by versions after a particular semester so we just queried by all versions. 
What I can do: 
Sort each version of the returned post requisite course by date.
For each version: 
      If this version of the post requisite has the given course number as a pre-requisite, Add the post requisite course to the post requisite field of version of the original course that occured before the post requisite version.
    Since we also need to store the pre-reqs of the post-reqs as a field, I will take the latest version of the course and its pre-reqs.


Note, its pretty hard to store a specific version of post requisite right now, because we calculate post-reqs based on pre-requisites, which is tracked by course numbers(ex: IP Spring 2023 has Gateways Course number as a pre req, and not specifically all version of Gateway before Spring 2023). This leads to some pretty pathological cases:
Suppose IP 2023 has Gateway as a pre-req but IP 2024 does not. Then, should Gateway 2021 store IP as a post-req or not? Personally, I think it all versions of Gateway should still just store IP as a post-req as we're using post-reqs more as a feature to show future courses students might be interested in, and regardless of Gateway being an IP pre-req, its still useful for students to know they should consider taking IP after Gateway.

The other solution to the above issue is for each version of a course, we also store the specific version of the course that is a post req. But, that leads to its own headaches in the frontend haha. Lmk if you want us to implement this. 

I think the changes you suggested makes sense, but were pretty complicated so we decided to try implementing version agnostically first and figured could come back and make this change if we have time (which is now lol).

Just to play devil's advocate as well, some thoughts on why doing post-reqs version agnostically could be better:
 The versioning system on uCredit is insanely buggy from what I've seen. Most of the time that I've tried it in prod, the wrong course version shows up when you click on a course.  So I wanted to make post-reqs version-agnostic since a courses pre-reqs very rarely change from what I saw in the database. This way, even if the wrong version loads on the frontend, you'll still see the right set of pre-reqs. 

Thank you for reading my ted talk :) haha, but long story short, I think I'll implement the the fix at the very top as that eliminates most bugs?

Reply:
Or wait actually, I think its not too complicated to do the second solution, it'd also make it easier to know which version of a post req course to display when we click on it?